File: src/main/java/net/kenro/ji/jin/purescript/psi/PSProperName.java
package net.kenro.ji.jin.purescript.psi;


import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiNameIdentifierOwner;

public interface PSProperName extends PsiNameIdentifierOwner {

    String getName();

    PsiElement setName(String newName);

    PsiElement getNameIdentifier();
}

File: src/main/java/org/purescript/psi/name/PSProperName.kt
package org.purescript.psi.name

import com.intellij.lang.ASTNode
import org.purescript.psi.PSPsiElement

class PSProperName(node: ASTNode) : PSPsiElement(node) {
    override fun getName(): String {
        return text.trim()
    }
}
